File: BasicAuthentication/ViewModels/Responses/Response.cs
using BasicAuthentication.ViewModels.Enums;

namespace BasicAuthentication.ViewModels.Responses;

public class Response
{
    /// <summary>
    /// 생성자
    /// </summary>
    /// <param name="result">로그인 결과</param>
    /// <param name="code">코드</param>
    /// <param name="message">메시지</param>
    public Response(EnumResponseResult result , string code, string message)
    {
        Result = result;
        Code = code;
        Message = message;
    }

    /// <summary>
    /// 생성자
    /// </summary>
    /// <param name="code">코드</param>
    /// <param name="message">메시지</param>
    public Response(string code , string message)
    {
        Code = code;
        Message = message;
    }

    /// <summary>
    /// 응답 메세지 
    /// </summary>
    public string Message { get; set; }

    /// <summary>
    /// 응답 코드 
    /// </summary>
    public string Code { get; set; }

    /// <summary>
    /// 응답 결과 
    /// </summary>
    public EnumResponseResult Result { get; set; } = EnumResponseResult.Error;
}

File: BudgetBringerV2/src/models/responses/response.ts
import {EnumResponseResult} from "../enums/enum-response-result";

/**
 * 응답 기본 데이터
 */
export class Response {
  /**
   * 응답 결과
   */
  public result : EnumResponseResult = EnumResponseResult.error;
  /**
   * 응답 메세지
   */
  public message: string;
  /**
   * 코드
   */
  public code : string;
  /**
   * 권한 여부
   */
  public isAuthenticated = false;
  /**
   * is Result Error
   */
  public error  = () => {
    return this.result === EnumResponseResult.error;
  }
  /**
   * is Result Success
   */
  public success  = () => {
    return this.result === EnumResponseResult.success;
  }
  /**
   * is Result Warning
   */
  public warning  = () => {
    return this.result === EnumResponseResult.waring;
  }
}
